build_modules/lib/src/errors.dart

```diff
@@ -102,3 +104,36 @@ Future<String> _missingImportMessage(
   var lineInfo = parsed.lineInfo.getLocation(import.offset);
   return '`$import` from $sourceId at $lineInfo';
 }
+
+/// An [Exception] that is thrown when there are some unsupported modules.
+class UnsupportedModulesException implements Exception {
+  final Set<Module> unsupportedModules;
+
+  UnsupportedModulesException(this.unsupportedModules);
+
+  Stream<ModuleLibrary> exactLibraries(AssetReader reader) async* {
+    for (var module in unsupportedModules) {
+      for (var source in module.sources) {
+        var libraryId = source.changeExtension(moduleLibraryExtension);
+        ModuleLibrary library;
+        if (await reader.canRead(libraryId)) {
+          library = ModuleLibrary.deserialize(
+              libraryId, await reader.readAsString(libraryId));
+        } else {
```

Comment:
why would we need this branch?

Reply:
For part files - we don't output a `.library` file. However i just realized we are skipping them anyways so I will change the branch to a `continue` instead.